### PR TITLE
OBSDOCS-1423: Document that instrumentation injection is GA

### PR DESCRIPTION
**Pull request by max-cx**
:warning: DO NOT MERGE! `do-not-merge/hold`

:warning: This PR is tied to a product release date, which is currently set to **December 10**.

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

:warning: No Change Management in this PR because the Technology Preview admonition is moved from the assembly per-page level to the module per-section level because of the sections is GA'd in https://github.com/openshift/openshift-docs/pull/84370.

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16, 4.17, 4.18

Issue:

https://issues.redhat.com/browse/OBSDOCS-1423

Link to docs preview:

https://84370--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-instrumentation.html#otel-configuration-of-opentelemetry-sdk-variables_otel-configuration-of-instrumentation

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-config-instrumentation.adoc [hunk 1]
```diff
@@ -98,9 +98,66 @@ spec:
 
 |===
 
+[id="otel-configuration-of-opentelemetry-sdk-variables_{context}"]
+== Configuration of the OpenTelemetry SDK variables
+
+You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resourceto instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:104]:
```suggestion
You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resource to instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:
```

**Review by stevsmit — COMMENTED** [review]
Mostly looks good. Be sure to fix the featurename issue and consider "not supported" vs. "unsupported."

File: modules/otel-config-instrumentation.adoc [hunk 2]
```diff
@@ -98,9 +98,45 @@ spec:
...
+* `OTEL_EXPORTER_OTLP_CERTIFICATE`
+* `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE`
+* `OTEL_EXPORTER_OTLP_CLIENT_KEY`
+
+.Values for the `instrumentation.opentelemetry.io/inject-sdk` annotation
+[cols="a, a"]
+|===
+| Value | Description
+
+|`"true"`
```
Comment (IshwarKanse) [inline modules/otel-config-instrumentation.adoc:121]: The instrumentation CR name is also considered in the annotation. 

**"true" - to inject Instrumentation resource with default name from the current namespace.**
"my-instrumentation" - to inject Instrumentation CR instance with name "my-instrumentation" in the current namespace.
"my-other-namespace/my-instrumentation" - to inject Instrumentation CR instance with name "my-instrumentation" from another namespace "my-other-namespace".
"false" - do not inject

File: modules/otel-config-instrumentation.adoc [hunk 3]
```diff
@@ -227,6 +277,13 @@ $ oc adm policy add-scc-to-user otel-go-instrumentation-scc -z <service_account>
 [id="otel-configuration-of-java-auto-instrumentation_{context}"]
 == Configuration of the Java auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:280]: Thank you so much for spotting this. Fixing.

File: modules/otel-config-instrumentation.adoc [hunk 4]
```diff
@@ -98,9 +98,45 @@ spec:
...
+
+|`"<namespace>/<instrumentation_name>"`
+|Specifies the name of the `+Instrumentation+` resource to inject from another namespace.
+
+|===
+
 [id="otel-configuration-of-apache-http-server-auto-instrumentation_{context}"]
 == Configuration of the Apache HTTP Server auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:138]:
```suggestion
:FeatureName: The Apache HTTP Server auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: modules/otel-config-instrumentation.adoc [hunk 5]
```diff
@@ -142,6 +178,13 @@ instrumentation.opentelemetry.io/inject-apache-httpd: "true"
 [id="otel-configuration-of-dotnet-auto-instrumentation_{context}"]
 == Configuration of the .NET auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:181]:
```suggestion
:FeatureName: The .NET auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: modules/otel-config-instrumentation.adoc [hunk 6]
```diff
@@ -169,6 +212,13 @@ instrumentation.opentelemetry.io/inject-dotnet: "true"
 [id="otel-configuration-of-go-auto-instrumentation_{context}"]
 == Configuration of the Go auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:215]:
```suggestion
:FeatureName: The Go auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: modules/otel-config-instrumentation.adoc (hunk 3 above)
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:280]:
```suggestion
:FeatureName: The Java auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: modules/otel-config-instrumentation.adoc [hunk 7]
```diff
@@ -252,6 +309,13 @@ instrumentation.opentelemetry.io/inject-java: "true"
 [id="otel-configuration-of-nodejs-auto-instrumentation_{context}"]
 == Configuration of the Node.js auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:312]:
```suggestion
:FeatureName: The Node.js auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: modules/otel-config-instrumentation.adoc [hunk 8]
```diff
@@ -280,6 +344,13 @@ The `+instrumentation.opentelemetry.io/otel-go-auto-target-exe+` annotation sets
 [id="otel-configuration-of-python-auto-instrumentation_{context}"]
 == Configuration of the Python auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:347]:
```suggestion
:FeatureName: The Python auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

File: observability/otel/otel-configuration-of-instrumentation.adoc [hunk 9]
```diff
@@ -7,7 +7,6 @@ include::_attributes/common-attributes.adoc[]
 toc::[]
 
 :FeatureName: OpenTelemetry instrumentation injection
```
Comment (max-cx) [inline observability/otel/otel-configuration-of-instrumentation.adoc:9]:
```suggestion
```

File: modules/otel-config-instrumentation.adoc [hunk 10]
```diff
@@ -142,6 +178,13 @@ instrumentation.opentelemetry.io/inject-apache-httpd: "true"
 [id="otel-configuration-of-dotnet-auto-instrumentation_{context}"]
 == Configuration of the .NET auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+This feature by default injects not supported upstream instrumentation libraries.
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:185]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

File: modules/otel-config-instrumentation.adoc [hunk 11]
```diff
@@ -227,6 +277,13 @@ $ oc adm policy add-scc-to-user otel-go-instrumentation-scc -z <service_account>
 [id="otel-configuration-of-java-auto-instrumentation_{context}"]
 == Configuration of the Java auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+This feature by default injects not supported upstream instrumentation libraries.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:284]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

File: modules/otel-config-instrumentation.adoc [hunk 12]
```diff
@@ -252,6 +309,13 @@ instrumentation.opentelemetry.io/inject-java: "true"
 [id="otel-configuration-of-nodejs-auto-instrumentation_{context}"]
 == Configuration of the Node.js auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+This feature by default injects not supported upstream instrumentation libraries.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:316]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

File: modules/otel-config-instrumentation.adoc [hunk 13]
```diff
@@ -280,6 +344,13 @@ The `+instrumentation.opentelemetry.io/otel-go-auto-target-exe+` annotation sets
 [id="otel-configuration-of-python-auto-instrumentation_{context}"]
 == Configuration of the Python auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+This feature by default injects not supported upstream instrumentation libraries.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:351]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

File: modules/otel-config-instrumentation.adoc [hunk 14]
```diff
@@ -169,6 +212,13 @@ instrumentation.opentelemetry.io/inject-dotnet: "true"
 [id="otel-configuration-of-go-auto-instrumentation_{context}"]
 == Configuration of the Go auto-instrumentation
 
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+This feature by default injects not supported upstream instrumentation libraries.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:219]:
```suggestion
By default, this feature injects unsupported, upstream instrumentation libraries.
```

File: modules/otel-config-instrumentation.adoc (hunk 2 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:121]: Thank you, Ishwar. It's on the next several lines.

File: modules/otel-config-instrumentation.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:222]: Thank you for the suggestion. Fixed.

File: modules/otel-config-instrumentation.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:288]: Thank you for the suggestion. Fixed.

File: modules/otel-config-instrumentation.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:321]: Thank you for the suggestion. Fixed.

File: modules/otel-config-instrumentation.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:357]: Thank you for the suggestion. Fixed.

File: modules/otel-config-instrumentation.adoc (hunk 10 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:187]: Applied.

File: modules/otel-config-instrumentation.adoc [hunk 15]
```diff
@@ -98,9 +98,46 @@ spec:
 
 |===
 
+[id="otel-configuration-of-opentelemetry-sdk-variables_{context}"]
+== Configuration of the OpenTelemetry SDK variables
+
+You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resource to instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:
```
Comment (pavolloffay) [inline modules/otel-config-instrumentation.adoc:104]: to inject some of the following OpenTelemetry SDK environment variables depending on the Instrumentation CR into your pod.


The reason being not all of these env vars are injected, e.g. if TLS is not specified the  OTEL_EXPORTER_OTLP_CERTIFICATE (and others) are not set

**Review by pavolloffay — COMMENTED** [review]
Note that should be merged as part of the RHOSDT 3.4 release

File: modules/otel-config-instrumentation.adoc (hunk 15 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc]: Thank you. Applied.

**Review by pavolloffay — APPROVED** [review]

File: modules/otel-config-instrumentation.adoc [hunk 16]
```diff
@@ -98,9 +98,46 @@ spec:
...
+.Values for the `instrumentation.opentelemetry.io/inject-sdk` annotation
+[cols="a, a"]
+|===
+| Value | Description
+
+|`"true"`
+|Injects the `+Instrumentation+` resource with the default name from the current namespace.
+
+|`"false"`
+|Injects no `+Instrumentation+` resource.
```
Comment (xenolinux) [inline modules/otel-config-instrumentation.adoc:125]: It means that this setting does not inject `+Instrumentation+`, right? 

Can it be revised something like that?

File: modules/otel-config-instrumentation.adoc [hunk 17]
```diff
@@ -169,6 +214,14 @@ instrumentation.opentelemetry.io/inject-dotnet: "true"
 [id="otel-configuration-of-go-auto-instrumentation_{context}"]
 == Configuration of the Go auto-instrumentation
 
+:FeatureName: The Go auto-instrumentation
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+By default, this feature injects unsupported, upstream instrumentation libraries.
```
Comment (xenolinux) [inline modules/otel-config-instrumentation.adoc:222]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

File: modules/otel-config-instrumentation.adoc [hunk 18]
```diff
@@ -227,6 +280,14 @@ $ oc adm policy add-scc-to-user otel-go-instrumentation-scc -z <service_account>
 [id="otel-configuration-of-java-auto-instrumentation_{context}"]
 == Configuration of the Java auto-instrumentation
 
+:FeatureName: The Java auto-instrumentation
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+By default, this feature injects unsupported, upstream instrumentation libraries.
```
Comment (xenolinux) [inline modules/otel-config-instrumentation.adoc:288]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

File: modules/otel-config-instrumentation.adoc [hunk 19]
```diff
@@ -252,6 +313,14 @@ instrumentation.opentelemetry.io/inject-java: "true"
 [id="otel-configuration-of-nodejs-auto-instrumentation_{context}"]
 == Configuration of the Node.js auto-instrumentation
 
+:FeatureName: The Node.js auto-instrumentation
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+By default, this feature injects unsupported, upstream instrumentation libraries.
```
Comment (xenolinux) [inline modules/otel-config-instrumentation.adoc:321]:
```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

File: modules/otel-config-instrumentation.adoc [hunk 20]
```diff
@@ -280,6 +349,14 @@ The `+instrumentation.opentelemetry.io/otel-go-auto-target-exe+` annotation sets
 [id="otel-configuration-of-python-auto-instrumentation_{context}"]
 == Configuration of the Python auto-instrumentation
 
+:FeatureName: The Python auto-instrumentation
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[IMPORTANT]
+====
+By default, this feature injects unsupported, upstream instrumentation libraries.
```
Comment (xenolinux) [inline modules/otel-config-instrumentation.adoc:357]: Same comment

```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

**Review by xenolinux — COMMENTED** [review]
One suggestion; one question -- Otherwise looks great!

Merge review done.

**Review by IshwarKanse — APPROVED** [review]

File: modules/otel-config-instrumentation.adoc (hunk 17 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:222]: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.

File: modules/otel-config-instrumentation.adoc (hunk 19 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:321]: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.

File: modules/otel-config-instrumentation.adoc (hunk 20 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:357]: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.

**Pull request merged**